callbacks.go

```diff
@@ -33,6 +33,8 @@ func (b *Bot) AddCallback(value string, c Callback) {
 func (b *Bot) CallbackLoop() {
 	for {
 		select {
+		case <-b.stop:
```

Comment:
When we're listening on a channel in multiple places for the purposes of a broadcast like this, we usually call `close(chan)`. This is because a send to a channel will only go to _1_ listener, so some listeners won't get notified when we send on the channel.

On the other hand, when we do a `close(chan)`, they will all receive. It might seem a little counterintuitive, but here's a good writeup: https://blog.golang.org/pipelines